.gitlab-ci.yml

```diff
@@ -448,6 +449,65 @@ check-polkadot-companion-status:
   script:
     - ./.maintain/gitlab/check_polkadot_companion_status.sh
 
+pr-docker-build:            &pr-docker-build
```

Comment:
The consideration here is ```publish-docker-substrate``` only runs for ```publish``` stage, If its enabled for every commit, the official docker repo could be possibly polluted. The extra docker image publish does not really cost. What do you think?

Reply:
We are publishing the new `parity/substrate:<ver-hash>` with every successful build on `master`. You can use those images.

Reply:
> We are publishing the new parity/substrate:<ver-hash> with every successful build on master

Just to make sure we are on the same page and I am not missing something: The image needs to be build for the commit of the pull request under test. Any `master` image is of no use here.

Reply:
okay, use another image for that, but then make sure that you:
- are using the [template](https://github.com/paritytech/substrate/pull/5793/files#diff-96edf7a6f008de9e928d04e1ae5e12a5R529)
- since you trigger build for every commit, make sure you do not trigger subkey build
- not releasing regular substrate docker image for every commit

Reply:
Bump. Please use the template or add the conditions to an existing docker build job.

Reply:
Sorry, i think i misunderstood the comment. Could you please do a suggested change here?

Reply:
please use [the template](https://github.com/paritytech/substrate/blob/1de6acabd4ffc48e0b6dd59d3966de972ec07dcb/.gitlab-ci.yml#L574) to build your docker image.

Reply:
I added ```*default-vars``` to it. Since it is not pushing to same docker repo, is there anything else that i need to add to the job?